tests/testthat/test-derive_locf_records.R

```diff
@@ -304,3 +304,302 @@ test_that("derive_locf_records Test 6: populate VISITNUM for LOCF records", {
+          all other variables in the new records", {
+  input <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~PARAMN, ~PARAMCD, ~PARAM, ~AVAL, ~AVISITN, ~AVISIT, ~ADY,
+    "TEST01", "01-701-1015", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", 51, 0, "BASELINE", 0,
+    "TEST01", "01-701-1015", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", 50, 2, "WEEK 2", 14,
+    "TEST01", "01-701-1015", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", 52, 4, "WEEK 4", 28,
+    "TEST01", "01-701-1015", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", 54, 6, "WEEK 6", 42,
+    "TEST01", "01-701-1015", 2, "SYSBP", "Systolic Blood Pressure (mmHg)", 121, 0, "BASELINE", 0,
+    "TEST01", "01-701-1015", 2, "SYSBP", "Systolic Blood Pressure (mmHg)", 121, 2, "WEEK 2", 14,
+    "TEST01", "01-701-1028", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", 79, 0, "BASELINE", 0,
+    "TEST01", "01-701-1028", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", 80, 2, "WEEK 2", 12,
+    "TEST01", "01-701-1028", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", NA, 4, "WEEK 4", 28,
+    "TEST01", "01-701-1028", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", NA, 6, "WEEK 6", 44,
+    "TEST01", "01-701-1028", 2, "SYSBP", "Systolic Blood Pressure (mmHg)", 130, 0, "BASELINE", 0
+  )
+
+  advs_expected_obsv <- tibble::tribble(
+    ~PARAMCD, ~PARAM, ~AVISITN, ~AVISIT, ~ADY,
+    "DIABP", "Diastolic Blood Pressure (mmHg)", 0, "BASELINE", 0,
+    "DIABP", "Diastolic Blood Pressure (mmHg)", 2, "WEEK 2", 14,
+    "DIABP", "Diastolic Blood Pressure (mmHg)", 4, "WEEK 4", 28,
+    "DIABP", "Diastolic Blood Pressure (mmHg)", 6, "WEEK 6", 42,
+    "SYSBP", "Systolic Blood Pressure (mmHg)", 0, "BASELINE", 0,
+    "SYSBP", "Systolic Blood Pressure (mmHg)", 2, "WEEK 2", 14
+  )
+
+
+  expected_output <- bind_rows(
+    input,
+    tibble::tribble(
+      ~STUDYID, ~USUBJID, ~PARAMN, ~PARAMCD, ~PARAM, ~AVAL, ~AVISITN, ~AVISIT, ~ADY,
+      "TEST01", "01-701-1028", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", 80, 4, "WEEK 4", 28,
+      "TEST01", "01-701-1028", 1, "DIABP", "Diastolic Blood Pressure (mmHg)", 80, 6, "WEEK 6", 44,
+      "TEST01", "01-701-1028", 2, "SYSBP", "Systolic Blood Pressure (mmHg)", 130, 2, "WEEK 2", 14
+    ) %>%
+      mutate(DTYPE = "LOCF")
+  )
+
+
+  actual_output <- derive_locf_records(
+    input,
+    dataset_ref = advs_expected_obsv,
+    analysis_var = AVAL,
+    by_vars = exprs(STUDYID, USUBJID, PARAM, PARAMCD),
+    id_vars_ref = exprs(STUDYID, USUBJID, PARAM, PARAMCD, AVISITN, AVISIT),
+    imputation = "update_add",
+    order = exprs(AVISITN, AVISIT, ADY),
+    keep_vars = exprs(PARAMN)
+  )
+
+
+  expect_dfs_equal(
+    base = expected_output,
+    compare = actual_output,
+    keys = c("STUDYID", "USUBJID", "PARAMCD", "PARAM", "AVISITN", "AVISIT", "DTYPE")
+  )
+})
+
+
+
+## Test 11: fill variables other than analysis_var for imputed records ----
+test_that("derive_locf_records Test 11: fill variables other than analysis_var for imputed
```

Comment:
I would combine Test 6 and Test 11 into a single test and update the title to state that it tests the `keep_vars` argument. Both tests test the `keep_vars` argument, they just test slightly different scenarios (Test 6: missing visit, Test 11: `AVAL` missing).